tests/node_eviction_test.go

```diff
@@ -0,0 +1,253 @@
+		output, err := ktests.RunCommand("oc", args...)
+		Expect(err).ToNot(HaveOccurred())
+
+		var count int
+		nodes := strings.Fields(output)
+		for _, node := range nodes {
+			args := []string{"get", "node", node}
+			output, err := ktests.RunCommand("oc", args...)
+			Expect(err).ToNot(HaveOccurred())
+
+			if !strings.Contains(output, "SchedulingDisabled") && !strings.Contains(output, "NotReady") {
+				count++
+			}
+		}
+		if count < 2 {
+			Skip("not enough schedulable nodes for this testing")
+		}
+
+	})
+
+	AfterEach(func() {
+		// nodes can be in drained status after some tests is failed, ensure uncordoning them after each tests.
+		uncordonNodes()
+	})
+
+	Describe("VirtualMachineInstance Evictions", func() {
+		BeforeEach(func() {
+			vmi.Manifest = "tests/manifests/vmi-ephemeral.yaml"
+			vmi.Name = "vmi-ephemeral"
+		})
+		It("virtualmachine instance will not be re-scheduled", func() {
+			By("Create the virtualmachine instance")
+			args := []string{"create", "-n", ns, "-f", vmi.Manifest}
+			_, err := ktests.RunCommand("oc", args...)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Check that the virtualmachine instance is running")
+			args = []string{"get", "-n", ns, "vmi", vmi.Name, "--template", "{{.status.phase}}"}
+			Eventually(func() string {
+				output, err := ktests.RunCommand("oc", args...)
+				Expect(err).ToNot(HaveOccurred())
+				return output
+			}, time.Minute*10).Should(Equal("Running"))
+
+			By("Retrieve the vmi's node")
+			args = []string{"get", "-n", ns, "vmi", vmi.Name, "--template", "{{.status.nodeName}}"}
+			nodeName, err := ktests.RunCommand("oc", args...)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Drain the vmi's node")
+			args = []string{"adm", "drain", nodeName, "--delete-local-data", "--ignore-daemonsets", "--force", "--pod-selector=kubevirt.io=virt-launcher"}
+			_, err = ktests.RunCommand("oc", args...)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Verify that the virtual machine instance is Failed")
+			args = []string{"get", "-n", ns, "vmi", vmi.Name, "--template", "{{.status.phase}}"}
+			output, err := ktests.RunCommand("oc", args...)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(output).To(Equal("Failed"))
+
+			By("Verify that the vmi's node is unschedulable")
+			args = []string{"get", "node", nodeName, "--template", "{{.spec.unschedulable}}"}
+			output, err = ktests.RunCommand("oc", args...)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(output).To(Equal("true"))
+
+			By("Uncordon the vmi's node")
+			args = []string{"adm", "uncordon", nodeName}
+			_, err = ktests.RunCommand("oc", args...)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Verify that the node is schedulable again")
+			args = []string{"get", "node", nodeName}
+			output, err = ktests.RunCommand("oc", args...)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(output).ToNot(ContainSubstring("SchedulingDisabled"))
+		})
+	})
+
+	PDescribe("VirtualMachineInstanceReplicaSet Eviction", func() {
```

Comment:
any special reason we use `PDescribe` here?

Reply:
Pending it because it's failed on d/s, will revert it when it's PASS on d/s.